File: 5kyu/moving_zeroes.py
#MOVING ZEROES TO THE END
# Write an algorithm that takes an array and moves all of the zeros to the end, preserving the order of the other elements.

# move_zeros([1, 0, 1, 2, 0, 1, 3]) # returns [1, 1, 2, 1, 3, 0, 0]

#MY SOLUTION
def move_zeros(lst):
    not_zero = list(filter(lambda x: x!=0, lst))
    zero = list(filter(lambda x: x==0, lst))
    not_zero = not_zero + zero
    return not_zero

#OTHER CLEVER SOLUTION
def move_zeros(arr):
    l = [i for i in arr if isinstance(i, bool) or i!=0]
    return l+[0]*(len(arr)-len(l))

File: 5kyu/movingZeroes.js
//MOVING ZEROES TO THE END
// Write an algorithm that takes an array and moves all of the zeros to the end, preserving the order of the other elements.

//moveZeros([false,1,0,1,2,0,1,3,"a"]) // returns[false,1,1,2,1,3,"a",0,0]

//MY SOLUTION
function moveZeroes(arr){
    let zeroes = []
    let notZeroes = []
    for(i = 0; i < arr.length; i++){
        if((arr[i] === 0) || (arr[i] === +0)){
            zeroes.push(arr[i]);
        }else{
            notZeroes.push(arr[i])
        }
    }
    notZeroes.push(...zeroes)
    return notZeroes
}

//OTHER CLEVER SOLUTION
var moveZeros = function (arr) {
    return arr.filter(function(x) {return x !== 0}).concat(arr.filter(function(x) {return x === 0;}));
}